site/docs/components/TransactionWrapper.tsx

```diff
@@ -0,0 +1,43 @@
+import { ReactNode } from 'react';
+import { erc20Abi } from 'viem';
+import { useAccount } from 'wagmi';
+import type { Abi, ContractFunctionName, Hex, Address } from 'viem';
+
+// TODO: move to types file
+type Contract = {
+  address: Address;
+  abi: Abi;
+  functionName: ContractFunctionName;
+  args?: { to: Hex; data?: Hex; value?: bigint }[];
+};
+
+type TransactionWrapperChildren = {
+  address: Address | undefined;
+  contracts?: Contract[];
+};
+
+type TransactionWrapperReact = {
+  children: (props: TransactionWrapperChildren) => ReactNode;
+};
+
+export default function TransactionWrapper({
+  children,
+}: TransactionWrapperReact) {
+  const { address } = useAccount();
+
+  // TODO: replace with actual contract
+  const contracts: Contract[] = [
+    {
+      address: '0xFBA3912Ca04dd458c843e2EE08967fC04f3579c2',
+      abi: erc20Abi,
+      functionName: 'approve',
+    },
```

Comment:
Now that I think about it, if we want to test a contract in our docs, probably that should be something super chill on Base Sepolia. Something to think about it on Monday.

Reply:
yeah agreed going to ask in the group for help on this